src/pages/Registration.tsx

```diff
@@ -0,0 +1,317 @@
+import {
+    Button,
+    Card,
+    CardActions,
+    CardContent,
+    Checkbox,
+    FormControl,
+    FormControlLabel,
+    Grid,
+    Link,
+    TextField,
+    Typography,
+} from '@mui/material';
+import Topbar from 'components/header/Topbar';
+import React, { useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+interface RegistrationRequest {
+    firstName: string;
+    lastName: string;
+    email: string;
+    company: string;
+    useCase: string;
+    acknowledgedDocuments: boolean;
+}
+
+interface Errors {
+    firstName: boolean;
+    lastName: boolean;
+    email: boolean;
+    company: boolean;
+    useCase: boolean;
+    acknowledgedDocuments: boolean;
+}
+
+const Registration = () => {
+    const intl = useIntl();
+
+    const [firstName, setFirstName] = useState<string>('');
+    const [lastName, setLastName] = useState<string>('');
+    const [email, setEmail] = useState<string>('');
+    const [company, setCompany] = useState<string>('');
+    const [useCase, setUseCase] = useState<string>('');
+    const [acknowledgedDocuments, setAcknowledgedDocuments] =
+        useState<boolean>(false);
+
+    const [errors, setErrors] = useState<Errors>({
+        firstName: false,
+        lastName: false,
+        email: false,
+        company: false,
+        useCase: false,
+        acknowledgedDocuments: false,
+    });
+
+    const handlers = {
+        updateTextInput:
+            (
+                controlName: keyof RegistrationRequest,
+                setControl: (value: React.SetStateAction<string>) => void
+            ) =>
+            (
+                event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+            ) => {
+                event.target.value = event.target.value.trimStart();
+
+                if (controlName === 'firstName' || controlName === 'lastName') {
+                    event.target.value = event.target.value.replace(
+                        /[0-9&^*|":<>{}`();@$#%+=?/[\]\\]/,
```

Comment:
Can we add a comment or store this regex in a const so I know what it's looking for LOL

Reply:
**PENDING:** I went back and forth on creating a constant for this expression. It was originally named `numericAndSpecialRegExp` as it prevents numeric and most special characters from being entered in the name controls. When the text control change handlers were consolidated and the expression was only used in one location, it made sense for me to remove the constant. However, regular expressions are not the easiest to digest, so I am more than happy to reintroduce the constant. Will update.